DigitalLearningSolutions.Web/Controllers/LearningPortalController/Current.cs

```diff
@@ -55,28 +55,39 @@ public async Task<IActionResult> AllCurrentItems()
         }
 
         [HttpPost]
+        [SetDlsSubApplication(nameof(DlsSubApplication.LearningPortal))]
         [Route("/LearningPortal/Current/CompleteBy/{id:int}")]
-        public IActionResult SetCurrentCourseCompleteByDate(int id, int day, int month, int year, int progressId)
+        public IActionResult SetCurrentCourseCompleteByDate(
+            int id,
+            int progressId,
+            EditCompleteByDateFormData formData
+        )
         {
-            if (day == 0 && month == 0 && year == 0)
+            if (progressId == 0)
+            {
+                return new StatusCodeResult(500);
+            }

```

Comment:
I thought I had a reason for this but now I can't remember what it was. Seems okay to remove it

Reply:
I suspect it was because this shares the view model with the new Action Plan editing, which doesn't need a progress ID, so this check makes sure that Courses go to the right action. But if the code is set up correctly, then anything that reaches this should already have a progress Id.